### PR TITLE
Detach should use each(), not this.map()

### PR DESCRIPTION
**Pull request by amccollum**
I don't see why detach shouldn't be using `each`. It's returning the same element set. Right now, code like the following will break:

```
$('div.some-div').detach().attr('some-attr')
```

Changing it to use `each` will fix this without needing to define a new method in the bridge.

### CODE REVIEW

**Comment by ded** [comment]
i see. makes sense
